### ISSUE TITLE
Statistics endpoint

### ISSUE THREAD

**Issue by erikgrinaker** [sanity-io/gradient#453]
I'd like a namespace statistics endpoint, only available internally (through e.g. JWT), which returns the number of documents, the number of distinct types, and the total document size, grouped by namespace. This should take a document filter, so that I can apply e.g. `deleted = FALSE AND id NOT LIKE '_.%'` (SQL syntax used as example).

We need this to be able to meter resource usage for billing in Sanity. We're currently fetching this directly from the Gradient database, which is ugly and brittle, using the following SQL queries:

```sql
-- Statistics for live documents
SELECT
  SPLIT_PART(namespaces.name, '.', 1) AS project,
  COUNT(*) AS documents,
  COUNT(DISTINCT documents.type) AS types,
  COUNT(DISTINCT SPLIT_PART(namespaces.name, '.', 2)) AS namespaces,
  COUNT(DISTINCT CASE WHEN ARRAY_LENGTH(STRING_TO_ARRAY(namespaces.name, '.'), 1) = 3 THEN namespaces.name END) AS datasets,
  SUM(pg_column_size(documents.attributes)) AS size
FROM
  documents,
  namespaces
WHERE
  documents.namespace_id = namespaces.id
  AND namespaces.name NOT IN ('null', 'sanity')
  AND documents.deleted = FALSE
  AND documents.id NOT LIKE '_.%'
GROUP BY
  project
ORDER BY
  project ASC;

-- Statistics for deleted documents
SELECT
  SPLIT_PART(namespaces.name, '.', 1) AS project,
  COUNT(*) AS deleted,
  SUM(pg_column_size(documents.attributes)) AS deleted_size
FROM
  documents,
  namespaces
WHERE
  documents.namespace_id = namespaces.id
  AND namespaces.name NOT IN ('null', 'sanity')
  AND documents.deleted = TRUE
  AND documents.id NOT LIKE '_.%'
GROUP BY
  project
ORDER BY
  project ASC;
```

**Comment by atombender** [comment]
Initial implementation in 5a99e7967e717e9d344b87e9336a88097e1c852a and de35f31086a4d0636d0dcae5622ce348a05741a6.

**Comment by erikgrinaker** [comment]
Thanks! Minor niggle: integers are returned as strings, e.g. `"count": "2093"`.

**Comment by atombender** [comment]
Ugh, that's in the JSON serializer library. I'll take a look.

**Comment by atombender** [comment]
Fixed in 210ffe2a.
